cachi2/core/package_managers/bundler/main.py

```diff
@@ -162,6 +167,27 @@ def _prepare_for_hermetic_build(source_dir: RootedPath, output_dir: RootedPath)
         BUNDLE_VERSION: "system"
     """
     )
+    # Note: if a package depends on a git revision then
+    # BUNDLE_DISABLE_LOCAL_BRANCH_CHECK is necessary for a hermetic build
```

Comment:
I don't think this is sufficient, I think we need `BUNDLE_DISABLE_LOCAL_REVISION_CHECK` as well. IIUC these checks are tied to whether the git dependency is going to be referenced with `ref:` AND/OR `branch:` in the Gemfile (which then somehow translates to Gemfile.lock) .